File: favorite-quote-backend/src/main/java/com/david/favoritequote/Service/QuoteService.java
package com.david.favoritequote.Service;

import com.david.favoritequote.Entity.Quote;
import org.springframework.http.ResponseEntity;

import java.util.List;

public interface QuoteService {


    List<Quote> getAllQuote();

    Quote getQuoteByID(int quoteID);

    ResponseEntity<Quote> addQuote(Quote quote, Integer categoryID);

    ResponseEntity<Quote> updateQuote(int quoteID, Quote quote);

    String deleteQuote(int quoteID);

    //get all post by category
    List<Quote> getQuoteByCategory(Integer categoryID);

    List<Quote> searchQuote(String keyword);
}

File: favorite-quote/src/services/quote-service.js
import axios from "axios";

const QUOTE_BASE_REST_API_URL = "http://localhost:9090/api/quotes/";

class QuoteService {
  getAllQuotes() {
    return axios.get(QUOTE_BASE_REST_API_URL);
  }

  // addQuote(quote){
  //     return axios.post(QUOTE_BASE_REST_API_URL, quote)
  // }

  addQuote(quote) {
    console.log(quote);
    return axios.post(
      `${QUOTE_BASE_REST_API_URL}category/${quote.categoryID}/`,
      quote
    );
  }

  getQuoteByID(quoteID) {
    return axios.get(`${QUOTE_BASE_REST_API_URL}`, quoteID);
  }

  updateQuote(quoteID, quote) {
    return axios.put(QUOTE_BASE_REST_API_URL + quoteID, quote);
  }

  deleteQuote(quoteID) {
    return axios.delete(QUOTE_BASE_REST_API_URL + quoteID);
  }
}

export default new QuoteService();
